### PR TITLE
Early networking: don't hard stop on configuration failure

### PR DESCRIPTION
**Pull request by internet-diglett**
Sometimes there are errors that can occur during the configuration of an individual link or peer that should not halt the process of starting up the entire rack. Instead, we should make an effort to configure these items, log an error if one is present, then move on. The worst case scenario is ultimately the same: in the event that we don't get enough of a working configuration to start the rack, NTP will not come up, and thus the control plane will never start. However now we can proceed in the event of a partial failure or misconfiguration and possibly get the control plane running, allowing users / operators to troubleshoot and update configurations via the api.

## Related

https://github.com/oxidecomputer/dendrite/issues/1048

### CODE REVIEW

File: sled-agent/src/bootstrap/early_networking.rs [hunk 1]
```diff
@@ -443,38 +445,36 @@ impl<'a> EarlyNetworkSetup<'a> {
...
+                        "dendrite not available, re-attempting port configuration in 5 seconds";
+                        "port_id" => ?port_id,
+                        "configuration" => ?dpd_port_settings,
+                    );
+                    sleep(Duration::from_secs(5)).await;
+                    continue;
+                } else {
+                    // log and move on to the next uplink instead of bailing on the
+                    // entire uplink process
+                    error!(
```
Comment (Nieuwejaar) [inline sled-agent/src/bootstrap/early_networking.rs:468]: It should probably still be an error if all of the uplinks fail.

File: sled-agent/src/bootstrap/early_networking.rs [hunk 2]
```diff
@@ -646,17 +655,22 @@ impl<'a> EarlyNetworkSetup<'a> {
...
-        mgd.static_add_v4_route(&rq).await.map_err(|e| {
-            EarlyNetworkSetupError::BgpConfigurationError(format!(
-                "static routing configuration failed: {e}",
-            ))
-        })?;
+
+        if let Err(e) = mgd.static_add_v4_route(&rq).await {
+            error!(
+                self.log,
+                "BGP peer configuration failed";
```
Comment (rcgoodfellow) [inline sled-agent/src/bootstrap/early_networking.rs:662]: This message should be about static routing, not BGP.

Comment (internet-diglett) [reply sled-agent/src/bootstrap/early_networking.rs:662]: Yay for copy paste errors

File: sled-agent/src/bootstrap/early_networking.rs (hunk 1 above)
Comment (andrewjstone) [reply sled-agent/src/bootstrap/early_networking.rs:468]: Good idea. Maybe maintain a count in the outer loop.

**Review by andrewjstone — APPROVED** [review]
Looks like it does what it says on the tin! Thanks @internet-diglett.

File: sled-agent/src/bootstrap/early_networking.rs (hunk 1 above)
Comment (internet-diglett) [reply sled-agent/src/bootstrap/early_networking.rs:470]: done

**Pull request merged**